src/qobj/operators.jl

```diff
@@ -11,6 +12,52 @@ export commutator
 export tunneling
 export qft
 
+@doc raw"""
+    rand_unitary(dimensions, distribution=:haar)
+
+Returns a random unitary [`QuantumObject`](@ref).
+
+The `dimensions` can be either the following types:
+- `dimensions::Int`: Number of basis states in the Hilbert space.
+- `dimensions::Vector{Int}`: list of dimensions representing the each number of basis in the subsystems.
+
+The `distribution` specifies which of the method used to obtain the unitary matrix:
+- `:haar`: Haar random unitary matrix using the algorithm from reference 1
+- `:exp`: Uses ``\exp(-iH)``, where ``H`` is a randomly generated Hermitian operator.
+
+# References
+1. [F. Mezzadri, How to generate random matrices from the classical compact groups, arXiv:math-ph/0609050 (2007)](https://arxiv.org/abs/math-ph/0609050)
+"""
+rand_unitary(dimensions::Int, distribution::Symbol = :haar) = rand_unitary([dimensions], Val(distribution))
+rand_unitary(dimensions::Vector{Int}, distribution::Symbol = :haar) = rand_unitary(dimensions, Val(distribution))
+function rand_unitary(dimensions::Vector{Int}, ::Val{:haar})
```

Comment:
I have never seen changing a method using a Val of a symbol. It is ok, but maybe we can use some structs instead? In this way we leave the same consistency we changing the method of some process, like wigner, spectrum, steadystate and any other. What do you think?